examples/polars/notebook.ipynb

```diff
@@ -0,0 +1,226 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "source": [
+    "Uncomment and run the cell below if you are in a Google Colab environment. It will:\n",
+    "1. Mount google drive. You will be asked to authenticate and give permissions.\n",
+    "2. Change directory to google drive.\n",
+    "3. Make a directory \"hamilton-tutorials\"\n",
+    "4. Change directory to it.\n",
+    "5. Clone this repository to your google drive\n",
+    "6. Move your current directory to the hello_world example\n",
+    "7. Install requirements.\n",
+    "\n",
+    "This means that any modifications will be saved, and you won't lose them if you close your browser."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "## 1. Mount google drive\n",
+    "# from google.colab import drive\n",
+    "# drive.mount('/content/drive')\n",
+    "## 2. Change directory to google drive.\n",
+    "# %cd /content/drive/MyDrive\n",
+    "## 3. Make a directory \"hamilton-tutorials\"\n",
+    "# !mkdir hamilton-tutorials\n",
+    "## 4. Change directory to it.\n",
+    "# %cd hamilton-tutorials\n",
+    "## 5. Clone this repository to your google drive\n",
+    "# !git clone https://github.com/DAGWorks-Inc/hamilton/\n",
+    "## 6. Move your current directory to the hello_world example\n",
+    "# %cd hamilton/examples/hello_world\n",
+    "## 7. Install requirements.\n",
+    "# %pip install -r requirements.txt\n",
+    "# clear_output()  # optionally clear outputs\n",
+    "# To check your current working directory you can type `!pwd` in a cell and run it."
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# Cell 2 - import modules to create part of the DAG from\n",
+    "# We use the autoreload extension that comes with ipython to automatically reload modules when\n",
+    "# the code in them changes.\n",
+    "\n",
+    "# import the jupyter extension\n",
+    "%load_ext autoreload\n",
+    "# set it to only reload the modules imported\n",
+    "%autoreload 1"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "import polars as pl\n",
+    "\n",
+    "from hamilton import (\n",
+    "    ad_hoc_utils,\n",
+    "    base,\n",
+    "    driver,\n",
+    ")\n",
+    "from hamilton.function_modifiers import extract_columns\n",
+    "from hamilton.plugins import h_polars"
+   ],
+   "metadata": {
+    "collapsed": false,
+    "pycharm": {
+     "name": "#%%\n"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# We'll place the spend calculations into a new module\n",
```

Comment:
@bryangalindo looks good. One edit please. Can you change this cell?

What you have works, however we want to minimize the use of adhoc_utils, since we don't want to proliferate that too much. It doesn't make it easy to go to production easily. 

Instead we either want to import `my_functions` directly (so delete this cell),  or use the `%%writefile my_functions.py` syntax at the top of the cell. In any case, the next cell would do `import my_functions` and pass that into the driver.  Thoughts? What would be more clear to a new person? Maybe the `%%writefile` option?

Reply:
@skrawcz thanks for the feedback! A couple of thoughts come to mind:
- Who are the primary consumers of the Jupyter Notebooks (e.g., data scientists)? 
- Would the primary consumers know about the `%%writefile` option? 
- Are there other Jupyter Notebooks with the same consumers that use the `%%writefile` option? [**1**]

Even if they wouldn't be familiar with it, I like that with the `%%writefile` option, we can still expose the e.g., spend functions so it's clear what we're sending into the driver (vs `import my_functions` -> `driver(..., my_functions, ...)`. 

So, I'm leaning more towards `%%writefile`. What are your thoughts?

[**1**] I can probably look this up myself, so no need to dig into it if you don't know off the top of your head.

Reply:
This should work
```python
%%writefile my_functions.py
# code goes here
```
Next cell
```python
# this should work because in you have loaded the auto reload extension in a prior cell.
%aimport my_functions
```
Next cell
```python
# driver code.
```